### PR TITLE
[Drop-In] Fix crash by removing fragment constructor

### PR DESCRIPTION
**Pull request by abhishek1508**
### Description

Fix crash as reported [here](https://github.com/mapbox/mapbox-navigation-android/pull/6618#discussion_r1026645961).

- Removed constructor arguments
- Added them as properties that are applied at runtime.

### CODE REVIEW

**Comment by abhishek1508** [comment]
@Zayankovsky Please take a look when you get a chance

File: libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/permission/PermissionsLauncherFragment.kt [hunk 1]
```diff
@@ -13,20 +13,19 @@ import androidx.fragment.app.FragmentActivity
...
  * A view-less fragment that requests permissions using [ActivityResultCaller] interface.
  */
-internal class PermissionsLauncherFragment(
-    private val permissions: Array<String>,
-    private val onResult: ActivityResultCallback<Map<String, Boolean>>
-) : Fragment() {
+internal class PermissionsLauncherFragment : Fragment() {
 
+    private var permissions: Array<String>? = null
+    private var onResult: ActivityResultCallback<Map<String, Boolean>>? = null
```
Comment (abhishek1508) [inline libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/permission/PermissionsLauncherFragment.kt:19]: I could have used `PermissionsLauncherFragment.newInstance(args)` and `setArguments` but given the `internal` and `private` nature of this fragment, I found this simple enough.

**Review by tomaszrybakiewicz — APPROVED** [review]
@abhishek1508 Thanks for patching this. Sorry. I completely forgot to test device rotation in my PR.
I hope you don't mind - I updated PermissionsLauncherFragment to avoid the use of force unwrapping and added CHANGELOG entry.

**Comment by Zayankovsky** [comment]
> @Zayankovsky Please take a look when you get a chance

The crash I mentioned [here](https://github.com/mapbox/mapbox-navigation-android/pull/6618#issuecomment-1320285573) is reproducible with this PR.

**Comment by abhishek1508** [comment]
> > @Zayankovsky Please take a look when you get a chance
> 
> The crash I mentioned [here](https://github.com/mapbox/mapbox-navigation-android/pull/6618#issuecomment-1320285573) is reproducible with this PR.

This should now be fixed @Zayankovsky

**Review by Zayankovsky — APPROVED** [review]

**Pull request merged**